where-to-travel/src/main/webapp/script.js

```diff
@@ -86,10 +91,41 @@ async function initialize() {
     styles: MAP_STYLES,
   };
   map = new google.maps.Map(document.getElementById('map'), mapOptions);
-
   showInfoModal();
 }
 
+/**
+ * Attaches listeners to the focusout event for search inputs.
+ */
+function attachSearchValidation() {
+  addListenerToSearchInput(document.getElementById(HOURS_ID), 'hour', HOURS_MAX_SEARCH);
+  addListenerToSearchInput(document.getElementById(MINUTES_ID), 'minute', MINUTE_MAX_SEARCH);
+}
+
+/**
+ * Adds an on focusout event to a dom element, which adds a formatted HTML tip to the DOM.
+ *
+ * @param element the dom element to add listener to
+ * @param type a string of the type of time input this element takes ('minute' or 'hour')
+ * @param max the maximum value of this input element
+ */
+function addListenerToSearchInput(element, type, max) {
+  element.addEventListener('focusout', function (event) {
+    $('#' + type + '-feedback').remove();
+    // if value is empty, set to 0, otherwise, parse the value
+    const stringInput = event.target.value;
+
+    const intValue = (stringInput === '') ? 0 : parseInt(stringInput);
+    if (!INT_REGEX_MATCHER.test(stringInput)) {
+      $('#' + FEEDBACK_ID).append('<p id="' + type +'-feedback" class="feedback">Please input a valid ' + type + ' integer without numbers or other characters</p>');
```

Comment:
Perhaps it might be better to say something along the lines of "Please input a valid whole number between 0 and max" and make the condition "if (!INT_REGEX_MATCHER || intValue < 0 || intValue > max)." 

I'm not entirely sure but i think that the average user might not remember what an integer is.

Reply:
Hm good idea. I also like chaining all of the statements, I will add this

Reply:
resolved in 7de47ae